torch/_dynamo/symbolic_convert.py

```diff
@@ -3328,19 +3351,28 @@ def STORE_GLOBAL(self, inst):
 
 class InliningGeneratorInstructionTranslator(InliningInstructionTranslator):
     generated_items: List[VariableTracker]
+    # Flag wether or not the InlineGenerator should consume the entire iterator
+    consume_all_items: bool
 
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self.generated_items = []
+        self.consume_all_items = True
+        self.generator_exhausted = False
 
     def YIELD_VALUE(self, inst: Instruction):
-        self.generated_items.append(self.pop())
+        top = self.pop()
+        self.generated_items.append(top)
         if len(self.generated_items) > MAX_ITERATOR_LIMIT:
             unimplemented(
                 "Too many yield values in generator. Maybe you are inlining an infinite generator. "
                 f"If not, please report a bug at {PT2_ISSUE_TRACKER_URL}",
             )
         self.push(ConstantVariable.create(None))
+        if not self.consume_all_items:
+            self.symbolic_result = top
+            # Stop tracing
+            raise YieldValueOp
```

Comment:
Trying to understand. The idea is 

* For context manager, we will set `consume_all_items` to be False. This means, any time we have a `yield ...` in the python code, we will effectively return to the parent frame (instead of tracing the rest of the frame). Is the correct?
* If that is the case, how does Dynamo know that it has to start from the next bytecode when the generator is called again? We are not saving the the equivalent of instruction pointer here.

cc @zou3519

Reply:
> Is the correct?

Yes, that's correct.

> If that is the case, how does Dynamo know that it has to start from the next bytecode when the generator is called again?

We create an inline tracer when `call_function` is called and reuse it when `next_variable` is called. The inline tracer saves (or holds) the state.

Reply:
Ah, clever!